sdk/turing/ensembler.py

```diff
@@ -87,18 +92,32 @@ def _ensemble_request(self, model_input: Dict[str, Any]) -> Any:
         Helper function to ensemble single requests; works on dictionary input in a single request made to the pyfunc
         ensembler service (run by the pyfunc ensembler service engine)
         """
+        request_body = model_input['body']
         # Get a mapping between route names and their corresponding responses
         routes_to_response = dict()
-        for prediction in model_input['response']['route_responses']:
+        for prediction in request_body['response']['route_responses']:
             routes_to_response[prediction["route"]] = prediction.copy()
             # Deletes route from the dictionary as it is a duplicate of the key
             del routes_to_response[prediction["route"]]["route"]
 
-        return self.ensemble(
-            input=model_input['request'],
-            predictions=routes_to_response,
-            treatment_config=model_input['response']['experiment']
-        )
+        try:
+            return self.ensemble(
+                input=request_body['request'],
+                predictions=routes_to_response,
+                treatment_config=request_body['response']['experiment'],
+                headers=model_input['headers']
+            )
+        except TypeError as e:
+            if "got an unexpected keyword argument 'headers'" in str(e):
+                # This handles the legacy ensemblers
+                # TODO: Deprecate support for legacy ensemblers
```

Comment:
Should perhaps specify what future SDK version would we be deprecating the legacy ensemblers.

Reply:
I think deprecating this depends on users moving away as well. So we need to do it at some point, but I'm not sure when. We will also need to remove some of the [default_route_id](https://github.com/gojek/turing/blob/main/sdk/turing/router/config/router_config.py#L135) related stuff. Will also track this in the backlog.